src/classes/frSubscription.cls

```diff
@@ -74,21 +74,83 @@ public class frSubscription extends frModel implements frSyncable {
         String name = String.valueOf(request.get('name'));
         String funraiseSupporterId = String.valueOf(request.get('supporterId'));
         List<Contact> contacts = [SELECT Id from Contact WHERE fr_Id__c = :funraiseSupporterId];
-        if(contacts.isEmpty()) {
+        List<Account> accounts;
+        if (frUtil.hasNPCobjects()) {
+            accounts = Database.query('SELECT Id, PersonContactId FROM Account WHERE fr_ID__c = :funraiseSupporterId');
+        }
+        if(contacts.isEmpty() && accounts.isEmpty()) {
             if(createLogRecord) frUtil.logRelationshipError(getFrType(), funraiseId, 
                                         frUtil.Entity.SUPPORTER, funraiseSupporterId);
             return result;
         }
-        
-        Subscription__c subscription = new Subscription__c(
-        	fr_ID__c = funraiseId,
-            Name = frUtil.truncateToFieldLength(
-                Subscription__c.Name.getDescribe(), 
+        Subscription__c subscription;
+        if(contacts.isEmpty() && frUtil.hasNPCobjects()) {
+            SObject gc = null;
+            List<SObject> gcList = Database.query('SELECT Id, Name FROM GiftCommitment WHERE fr_ID__c = :funraiseId');
+            if(gcList.isEmpty()) {
+                gc = Schema.getGlobalDescribe().get('giftcommitment').newSObject();
+            } 
+            else {
+                gc = gcList.get(0);
+            }
+            gc.put('fr_ID__c', funraiseId);
+            gc.put('Name', frUtil.truncateToFieldLength(
+                Schema.getGlobalDescribe().get('GiftCommitment').getDescribe().fields.getMap().get('Name').getDescribe(), 
                 String.valueOf(request.get('name'))
-            ), 
-            Supporter__c = contacts.get(0).Id
-        );
-        
+            ));
+            gc.put('DonorId', accounts.get(0).Id);
+
+            // campaign check for gc
+            String goalId = String.valueOf(request.get('campaignGoalId'));
+            if (String.isNotBlank(goalId)) {
+                List<Campaign> campaigns = [SELECT Id FROM Campaign WHERE fr_ID__c = :goalId];
+                if (campaigns.size() > 0) {
+                    gc.put('CampaignId', campaigns.get(0).Id);
+                } else {
+                    if(createLogRecord) frUtil.logRelationshipError(getFrType(), funraiseId, 
+                                                frUtil.Entity.CAMPAIGN, goalId);
+                }
+            }
+
+            String strPledgeAmount2 = String.valueOf(request.get('pledgeAmount'));
```

Comment:
Can we fix the variable naming here, I'm not following why this is called strPledgeAmount2

Reply:
Fixed!